b2sdk/transfer/emerge/emerger.py

```diff
@@ -63,6 +63,12 @@ def emerge(
         """
         planner = self.get_emerge_planner(recommended_upload_part_size=recommended_upload_part_size)
         emerge_plan = planner.get_emerge_plan(write_intents)
+        if emerge_plan.is_large_file():
+            [write_intent] = write_intents
```

Comment:
there are multiple issues with this line:
1) `write_intents` suggests that it can contain multiple write intents - most usages in `Bucket` have single intent, but `Emerger` can handle multiple write intents - this is also documented clearly in doc string
2) `write_intents` can be an iterator - doc string says "iterable" - and if it is iterator - then `get_emerge_plan` is reading it to the end - and this line would fail